velox/serializers/PrestoSerializer.cpp

```diff
@@ -641,30 +594,34 @@ void readRowVector(
     }
   }
 
-  readNulls(source, size, reused);
+  readNulls(source, size, *result);
 
-  // if offsets is needed, reconstruct aligned layout.
+  // If offsets is needed, reconstruct aligned layout.
   // TODO: make it efficient
-  if (needOffsets) {
-    tempChildren.resize(numChildren);
-    auto sourceRow = dynamic_cast<RowVector*>((*result).get());
-    for (int32_t child = 0; child < numChildren; ++child) {
-      tempChildren[child] = BaseVector::create(childTypes[child], size, pool);
-      auto src = sourceRow->childAt(child);
-      for (int32_t i = 0; i < size; ++i) {
-        if (!sourceRow->isNullAt(i)) {
-          tempChildren[child]->copy(src.get(), i, rawOffsets[i], 1);
-        }
+  if (!needOffsets) {
+    return;
+  }
+
+  std::vector<VectorPtr> tempChildren;
+  tempChildren.resize(numChildren);
+  auto sourceRow = dynamic_cast<RowVector*>(result.get());
+  for (int32_t child = 0; child < numChildren; ++child) {
+    tempChildren[child] = BaseVector::create(childTypes[child], size, pool);
+    auto& src = sourceRow->childAt(child);
```

Comment:
Are we guaranteed children will be non nulL ?

Reply:
I think so.